Enterprise-CMCS/cmcs-eregulations
# EREGCSC-2374 -- Alphabetize subject chips for each item

Resolves [EREGCSC-2374](https://jiraent.cms.gov/browse/EREGCSC-2374)

**Description**

Subject chips in result item need to be alphabetized based on our current display name hierarchy (short name > abbreviation > full name).

This is a front-end implementation because it is a very light lift to implement it this way.

**This pull request changes:**

- Adds `sortSubjects` method back into the filter methods on the front end
- Adds associated unit test and e2e test

**Steps to manually verify this change:**

1. Vitest tests pass
2. Cypress tests pass
3. Visit [experimental deployment](https://dj1158hrug.execute-api.us-east-1.amazonaws.com/dev1088) and add some docs with subjects and verify the ordering works as expected
